### PR TITLE
DC-3242 - return empty inbox when auth enrolments do not match enrolment params

### PR DESCRIPTION
**Pull request by elabeca**

### CODE REVIEW

**Review by alexflav23 — APPROVED** [review]

File: app/uk/gov/hmrc/securemessage/repository/ConversationRepository.scala [hunk 1]
```diff
@@ -73,26 +74,34 @@ class ConversationRepository @Inject()(implicit connector: MongoConnector)
     import uk.gov.hmrc.securemessage.models.core.Conversation.conversationFormat
 
     val querySelector = (enrolments, tags) match {
-      case (enrolments, None) => enrolmentQuery(enrolments)
-      case (enrolments, Some(tags)) if tags.nonEmpty =>
+      case (enrolments, _) if enrolments.isEmpty =>
+        JsObject.empty
+      case (enrolments, None) =>
+        enrolmentQuery(enrolments)
+      case (enrolments, Some(List())) =>
```
Comment (alexflav23) [inline app/uk/gov/hmrc/securemessage/repository/ConversationRepository.scala:81]: Could use `Nil` I believe

File: test/uk/gov/hmrc/securemessage/repository/ConversationRepositorySpec.scala [hunk 2]
```diff
@@ -106,22 +106,19 @@ class ConversationRepositorySpec extends PlaySpec with MongoSpecSupport with Bef
...
     }
 
-    "not be returned and instead a database exception raised when a tag filter without an enrolment counterpart is provided" in {
+    "none returned when a tag filter without an enrolment counterpart is provided" in {
       repoSetup()
-      an[reactivemongo.core.errors.DetailedDatabaseException] should be thrownBy {
+      val result =
         await(repository.getConversationsFiltered(Set.empty, Some(List(Tag("notificationType", "CDS Exports")))))
-      }
+      result.size mustBe 0
```
Comment (alexflav23) [inline test/uk/gov/hmrc/securemessage/repository/ConversationRepositorySpec.scala:113]: you can also use `result mustBe empty`

Comment (elabeca) [reply test/uk/gov/hmrc/securemessage/repository/ConversationRepositorySpec.scala:113]: get wartremover complaints [wartremover:Any] Inferred type containing Any: [e]scala.collection.GenTraversable[e]

Comment (elabeca) [reply test/uk/gov/hmrc/securemessage/repository/ConversationRepositorySpec.scala:113]: result mustBe Nil works

File: app/uk/gov/hmrc/securemessage/repository/ConversationRepository.scala (hunk 1 above)
Comment (elabeca) [reply app/uk/gov/hmrc/securemessage/repository/ConversationRepository.scala]: thanks

**Review by muralishub — APPROVED** [review]

**Pull request merged**